marshmallow/marshalling.py

```diff
@@ -231,7 +232,11 @@ def deserialize(self, data, fields_dict, many=False, partial=False,
         # Reset errors if not deserializing a collection
         if not self._pending:
             self.reset_errors()
-        if many and data is not None:
+        if many and data is not None and not isinstance(data, (list, set)):
```

Comment:
What about using marshmallow.utils.is_collection instead of this direct call to isinstance?